### PR TITLE
Feature/add laravel5.8 support

### PR DESCRIPTION
**Pull request by jimmypuckett**

### CODE REVIEW

**Comment by jimmypuckett** [comment]
@ssfinney I think that this is ready to review & tag as 2.0.0

File: .travis.yml [hunk 1]
```diff
@@ -1,41 +1,43 @@
...
-    - php: 7.1
-      env: setup=stable
     - php: 7.2
-      env: setup=lowest
     - php: 7.2
-      env: setup=stable
+      env: SETUP=lowest
+    - php: 7.3
+    - php: 7.3
+      env: SETUP=lowest
```
Comment (ssfinney) [inline .travis.yml:15]: Wait, this seems odd. Why are PHP 7.2 and 7.3 both marked as "SETUP=lowest"? Seems like one should be "stable" and one is "lowest".

Also, I don't see why there should be more than one version of 7.2 or 7.3.

File: .travis.yml [hunk 2]
```diff
@@ -1,41 +1,43 @@
...
   directories:
     - $HOME/.composer/cache
 
+services:
+  - memcached
+  - redis-server
+  - mysql
+
 before_install:
+  - phpenv config-rm xdebug.ini || true
```
Comment (ssfinney) [inline .travis.yml:29]: I'm confused; this always evaluates to "true"

File: .travis.yml [hunk 3]
```diff
@@ -1,41 +1,43 @@
...
+  - mysql -e 'CREATE DATABASE forge;'
 
 install:
-  - if [[ $setup = 'basic' ]]; then travis_retry composer install --no-interaction --prefer-source; fi
-  - if [[ $setup = 'stable' ]]; then travis_retry composer update --prefer-source --no-interaction --prefer-stable; fi
-  - if [[ $setup = 'lowest' ]]; then travis_retry composer update --prefer-source --no-interaction --prefer-lowest --prefer-stable; fi
-
-before_script: mkdir -p build/logs
+  - if [[ $SETUP = 'stable' ]]; then travis_retry composer update --prefer-dist --no-interaction --prefer-stable --no-suggest; fi
+  - if [[ $SETUP = 'lowest' ]]; then travis_retry composer update --prefer-dist --no-interaction --prefer-lowest --prefer-stable --no-suggest; fi
```
Comment (ssfinney) [inline .travis.yml:37]: Why "--prefer-lowest" and "--prefer-stable"? I would think for this one you would only want to use "--prefer-lowest"

**Review by ssfinney — COMMENTED** [review]
I've made some comments on some stuff I missed on the last PR (for the browser filter package)

**Comment by ssfinney** [comment]
Coveralls is having the same issue here.

File: .travis.yml (hunk 3 above)
Comment (jimmypuckett) [reply .travis.yml:37]: This is essentially the exact travis config form Laravel 5.8, except versions of php & dealing with code coverage.

What is happening, is that the `SETUP` env is defaulted to `stable` on line 5, so it runs composer with whatever the latest stable versions are. Then `SETUP` is changed to `lowest`, which pulls in the oldest possible version of packages, so it is a 2x2 matrix.

File: .travis.yml (hunk 2 above)
Comment (jimmypuckett) [reply .travis.yml:29]: Effectively, don't care if is error, but this is from Laravel's travis config.

File: .travis.yml (hunk 1 above)
Comment (jimmypuckett) [reply .travis.yml:15]: See other comment.

**Comment by jimmypuckett** [comment]
> Coveralls is having the same issue here.

Ya, I have no idea how to remove it.

**Pull request merged**